asynctest/mock.py

```diff
@@ -978,43 +978,54 @@ def __init__(self, gen, patchings, is_coroutine):
         self.__name__ = getattr(gen, '__name__', None)
         self.__qualname__ = getattr(gen, '__qualname__', None)
         self.patchings = patchings
+        self.global_patchings = [p for p in patchings if p.scope == GLOBAL]
+        self.limited_patchings = [p for p in patchings if p.scope == LIMITED]
 
         # GLOBAL patches have been started in the _patch/patched() wrapper
 
+    def _limited_patchings_stack(self):
+        with contextlib.ExitStack() as stack:
+            for patching in self.limited_patchings:
+                stack.enter_context(patching)
+
+            return stack.pop_all()
```

Comment:
I think you can also do this, which seems sligthly simpler:

    stack = contextlib.ExitStack()
    for patching in self._limited_patchings:
    stack.enter_context(patching)
    return stack

Reply:
There is a difference: currently if `enter_context` (i.e. `patching.__enter__`) raises an exception, the stack will properly close. The proposed change will lack this.

See https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack.pop_all

Reply:
You're right :)